### PR TITLE
Add target variables for 'enter-time' task

### PR DESCRIPTION
**Pull request by tudor-berariu**
Add a few more target vars.

### CODE REVIEW

**Review by mcobzarenco — APPROVED** [review]

**Comment by ppasupat** [comment]
Hi! Let me add a reviewer that can approve the pull request.

File: miniwob/html/miniwob/enter-text-2.html [hunk 1]
```diff
@@ -31,15 +33,17 @@
...
   } else {
     var expectedQ = q.toLowerCase();
+    TARGET_TEXT = expectedQ;
     q = q.toUpperCase();
     var queryText = 'all lower case letters';
   }
-  return {query: q, queryText: queryText, expectedText: expectedQ};
-}
+   return {query: q, queryText: queryText, expectedText: expectedQ};
+ }
```
Comment (ppasupat) [inline miniwob/html/miniwob/enter-text-2.html:46]: Nit: Dedent these two lines.

File: miniwob/html/miniwob/enter-text-ocr.html [hunk 2]
```diff
@@ -0,0 +1,104 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Find Word Task</title>
```
Comment (ppasupat) [inline miniwob/html/miniwob/enter-text-ocr.html:4]: Nit: Update the title.

File: miniwob/html/miniwob/read-table-2.html [hunk 3]
```diff
@@ -21,10 +21,12 @@
 <script src="../core/d3.v3.min.js"></script>
 <script src="../common/ui_utils.js"></script>
 <script>
+var TARGET_TEXT;
+
 core.EPISODE_MAX_TIME = 20000; // set episode interval to 20 seconds
 
 var genYear = function() {
-  return core.randi(1940, 2016).toString();
+   return core.randi(1940, 2016).toString();
```
Comment (ppasupat) [inline miniwob/html/miniwob/read-table-2.html:29]: Nit: Dedent the line.

**Comment by ppasupat** [comment]
@pseudo-rnd-thoughts This pull request edits the task HTML files by exposing task parameters as global JavaScript variables. This does not change the task behavior so it should be safe to merge. (Our Python API does not do anything with these variables, though.)

**Comment by mcobzarenco** [comment]
Oops, we've made a mistake, we didn't mean to create the pull request here yet, but instead into our fork. Also, btw amazing job on maintaining the MiniWoB++ repo!

But given we did create the PR here, a bit of context. We have written policies by hand (using [Playwright](https://playwright.dev/python/)) to solve all the MiniWoB++ tasks by only interacting with the environment at pixel level (e.g. actions are `Move(x, y)`, `LeftClick`, `EmitText("Hello")` etc.). An example:



You can use this to train an agent in a supervised fashion end-to-end or initialize an policies with "behaviour cloning" for RL.
---

To do this, we found it easier to add the ground truth to JS variables for some tasks which we read from the env -- hence what's going on in this PR. If you are open to merging this upstream, we'd be very happy to clean up this PR a bit and resubmit it :bow:

File: miniwob/html/miniwob/enter-text-ocr.html (hunk 2 above)
Comment (mcobzarenco) [reply miniwob/html/miniwob/enter-text-ocr.html:4]: :+1: -- not sure if we should include this at all, at least not in the same PR that puts the ground truth in JS variables.This `enter-text-ocr` is an additional task that requires entering text to help models learn OCR.

File: miniwob/html/miniwob/chase-circle.html [hunk 4]
```diff
@@ -21,7 +21,7 @@
 // eventually ends the episode.
 var translateCircle = function(translates) {
   if(translates !== 0){
-    d3.select('#circ').transition().duration(2000)
```
Comment (mcobzarenco) [inline miniwob/html/miniwob/chase-circle.html:24]: These changes to `chase-circle` should also be removed

**Review by mcobzarenco — CHANGES_REQUESTED** [review]

**Comment by pseudo-rnd-thoughts** [comment]
@ppasupat Thanks for confirming, the PR is all your choice of what we do include, or not, etc, you know better about the project than I do

**Comment by ppasupat** [comment]
@mcobzarenco Thank you for the context. Since this PR is intended to be for a fork, would it be OK if we close the PR for now?

Adding ground truth to JS variables is a great idea though, and we would gladly accept an implementation. One possible improvement would be to make this exposed information uniform across all tasks. The Python API would also benefit from this (currently the `fields` in the observation are extracted using regular expression, which is not ideal).

**Pull request closed without merging**